将AI审核功能添加到地下室预约流程中

Morgen-Kornblume opened the pull request:
- 地下室预约事由将可以由AI进行审核
- 目前仅支持 Ollama API 和 GLM API 的调用
- config 更新了 AI 审核功能控制，默认为关闭

Morgen-Kornblume committed:
添加了基本的config文件示例

Morgen-Kornblume committed:
未完成的AI审核功能接口

Morgen-Kornblume committed:
将预约事项审核插入地下室预约的流程

KeZhao783 committed:
feat(ai-inspection): 接入HTTP审核API，返回合规/不合规并附带理由

KeZhao783 committed:
Update AI_Inspection.py

KeZhao783 committed:
Update config.example.json

Morgen-Kornblume committed:
加入了ollama审核方法

Morgen-Kornblume committed:
Merge branch 'AI-Inspection' into AI-Inspection

Morgen-Kornblume committed:
Merge pull request #1 from KeZhao783/AI-Inspection

feat(ai-inspection): 接入HTTP审核API，返回合规/不合规并附带理由

Morgen-Kornblume committed:
整合了 ollama 方法和 GLM 方法

整合了 ollama 方法和 GLM 方法，并整合了相应的 config

Morgen-Kornblume committed:
优化了用户体验和提示措辞

Morgen-Kornblume committed:
取消对docker-compose.yml的修改

Deophius committed:
Merge branch 'develop' into AI-Inspection

Deophius commented on Appointment/utils/AI_Inspection.py:
```diff
@@ -0,0 +1,132 @@
+)
+
+
+def _call_glm_decision(text: str, timeout: int = 30) -> tuple[str, str]:
```
要不要考虑加一个日志，记录一下每次 API 调用用的 tokens 之类的，防止哪一天突然发现没钱了还不知道为什么

Morgen-Kornblume replied:
目前开发的是基于免费API的

Deophius commented on Appointment/utils/AI_Inspection.py:
```diff
@@ -0,0 +1,132 @@
+    # GLM 审核
+    # 组装要审核的文本
+    content = f"房间：{room_name}\n事由：{reason}"
+    # 超时时间可从配置读取，默认30秒
```
这里疑似还没有移到配置文件里

Deophius commented on Appointment/utils/AI_Inspection.py:
```diff
@@ -0,0 +1,132 @@
+            CONFIG.Ollama_ADDR + "/api/generate",
+            json={
+                "model": "gpt-oss:20b",
+                "prompt": f"你是一个审查学生预约功能房间信息的老师，请审核以下预约信息，审核要求是预约事由和房间用途匹配，且不能包含政治敏感信息和色情内容。房间名称：{room_name}。事由：{reason}。如果审核通过，只输出 1，不通过，则输出 0，然后接一个空格，在空格后输出不通过的原因，原因尽量简要；如无法判断，默认不通过",
```
这一段和 `SYSTEM_PROMPT` 是故意不一样的吗？

Morgen-Kornblume replied:
SYSTEM_PROMPT 是给 GLM 使用的，Ollama 的 GPT-OSS:20b 又另有一次文本组装。

Deophius commented on Appointment/utils/AI_Inspection.py:
```diff
@@ -0,0 +1,132 @@
+        else:
+            return False, f"房间用途不合规：{why or '无具体理由'}"
+    except Exception as e:
+        # API 调用失败：不通过并附带错误信息
```
个人认为 API 调用失败最好写到日志文件里，可能可以考虑加个微信提醒 (debug_stuid 那个)

Deophius commented on Appointment/views.py:
```diff
@@ -420,15 +420,16 @@ def arrange_time(request: HttpRequest):
-    max_appoint_time = int(CONFIG.max_appoint_time.total_seconds() // 3600) # 以小时计算
+    allow_overlap = CONFIG.allow_overlap
+    max_appoint_time = int(
+        CONFIG.max_appoint_time.total_seconds() // 3600)  # 以小时计算
```
怎么感觉 format 之后变丑了

Morgen-Kornblume replied:
不懂，疑似是pylance背着我们自己操作起来了

Deophius reviewed (commented):
是不是还需要考虑被用户玩坏的可能性，搞个 rate limiting 之类的

Morgen-Kornblume committed:
修正了遇到的一系列问题

- config_template.json 中添加了 GLM API 超时设定选项
- 新建了数据库中 AI 审核历史记录条目，和相应管理页面条目，可供反查/记录用户之前调用的时间和频次，防止刷接口
- 新增了API出现错误时记入日志以及微信通知 debug_stus 的功能（微信部分待测试）
AI 安全部分仍有待于更加广泛的和持久的测试，尽量减少被人喂毒输出奇奇怪怪东西的可能性

Morgen-Kornblume committed:
Merge branch 'develop' into AI-Inspection
